test/cpp/api/tensor.cpp

```diff
@@ -406,3 +406,31 @@ TEST(TensorTest, Requires_grad_) {
   ASSERT_THROWS_WITH(y.requires_grad_(false),
     "requires_grad_ is not implemented for Tensor");
 }
+
+TEST(TensorTest, Detach) {
+  auto x = torch::tensor({5}, at::TensorOptions().requires_grad(true));
+  auto y = x * x;
+  const auto y_detached = y.detach();
+  ASSERT_FALSE(y.is_leaf());
+  ASSERT_TRUE(y_detached.is_leaf());
```

Comment:
We might want to check `y_detached.requires_grad() == false`, based on https://pytorch.org/docs/stable/tensors.html#torch.Tensor.detach `The result will never require gradient`.